### DISCUSSION TITLE
Navigation display remaining + arrival time

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#492] [Q&A] [answered]
Allow to display both remaining + arrival time in navigation.

The default displays one panel that we can tap to switch between remaining and arrival time.

<img src="https://github.com/devemux86/cruiser/assets/3484020/9a8c7c0f-bfbe-41ac-ba8f-8ccf7907b71b" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/488fde58-65f9-4fc0-92f2-531d2a6e9fa3" width="50%"/>

**Comment by devemux86** [comment]
If we enable the panels for destination + waypoint and remaining + arrival time, they are 2x3 panels!
How should we handle this? 

Not allow remaining + arrival time panels when display destination + waypoint?
And instead, tap the time panels to switch between remaining and arrival time.

<img src="https://github.com/devemux86/cruiser/assets/3484020/e9a2db87-0848-4198-bd78-4b754bb8c328" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/48141dfd-9bcf-448d-8c3e-bc62c1679d18" width="50%"/>

**Reply by DIrgi** [reply]
I don't understand the question.
Leave it like it is + add the possibility to show all 4 data cells at the same time. 
This has to be done in the setup.  The limitation of displaying values is just the screen size.

**Comment by MotoUKRider** [comment]
I had wondered about what time panels to display - so I went with "Time Remaining" and rely on the Via-Point bar to tell me how far away the next stop is.  The time I arrive is dynamically calculated via optical means and mental arithmetic.  I try not use my fingers... :laughing: 

In all seriousness, being able to tap+toggle the time panel was enough for me and distance to Via-Point was also enough "detail".  It gets to a point where there's too much clutter and/or it gets a bit silly trying to work out what information panel you want to read while riding.

**Reply by WolfgangZ42** [reply]
> tap+toggle the time panel was enough for me and distance to Via-Point was also enough

same here

**Comment by devemux86** [comment]
A simple improvement is moving the waypoint distance / time / eta panels to the left.
And keep the destination distance / time / eta panels to the right.

So they are not mixed and it is much easier to quickly read them:
left is waypoint, right is destination.

**Reply by WolfgangZ42** [reply]
But **_please!_** keep it toggled - so that minimalists can use only 2 panels on the right side for every information.

**Reply by devemux86** [reply]
My comment was only for when we select in settings to display multiple panels.

The single panels - with data toggled via tap - remain as the defaults like now.

**Reply by bios4** [reply]
To be honest, I'm not sure if moving "some" panels to the left, while keeping others on the right is the best approach...
What's the benefit of filling up screen space on the left, just to keep it cleaner on the right!?

IF you add/implement the new/additional panels, I would suggest to let the user decide about the position (left or right).
Like it is already selectable for nearby POI/favorites.

**Reply by DIrgi** [reply]
It‘s all about activating/deactivating (when deactivated it’s not visible) Some like to have almost no infos on the screen, some, more - so finally fine for all. I like bios proposal if it’s from programmers perspective not to much work.

**Reply by devemux86** [reply]
We may add placement for all panels in the future. :slightly_smiling_face: 

Waypoint panels only appear _together_ with the destination panels, not by themselves.
So they are different from the nearby favorites and POI.

For example vs the above [image](https://github.com/devemux86/cruiser/discussions/492#discussioncomment-9086063) (where it was difficult to understand quickly distance / time).

<img src="https://github.com/devemux86/cruiser/assets/3484020/8e3cbca6-03b5-455b-8993-8db9aa11267e" width="50%"/>

**Reply by bios4** [reply]
> For example vs the above image

But this is still inconsistent!

Why is the distance (time) to the next waypoint shown on the left, while the distance (time) to the final destination is on the right?

When I want to check data for the NEXT waypoint: look left
When I want to check data for the NEXT nearby POI: look either left or right (depending on my setting)
When I want to check data for the NEXT nearby favorite: look either left or right (depending on my setting)
When I want to check data for the LAST waypoint/final destination: look right

This does not make too much sense TO ME without the option to decide for ALL panels where to be displayed.
Even causes confusion...

**Reply by MotoUKRider** [reply]
Actually, I quite like this example: My immediate (and maybe 2nd) turn/s on the left with my current direction plus Waypoint info.  The overall/master panels on the right.

I think "muscle memory" is the problem with most changes like this.

**Reply by devemux86** [reply]
Why we allow favorites or POI to the left? Then should we always have everything to the right in a long list?

It is easy to check quickly: waypoint info to the left and destination to the right.

Left image is unreadable unless stop driving and trying to understand the symbols.
At least the right image is clearly readable.

As already mentioned specific placement needs a lot of work and may come in the future.

<img src="https://github.com/devemux86/cruiser/assets/3484020/e9a2db87-0848-4198-bd78-4b754bb8c328" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/8e3cbca6-03b5-455b-8993-8db9aa11267e" width="50%"/>

**Reply by bios4** [reply]
> At least the right image is clearly readable.

But not only because of the left/right change, but also because at least two items are missing at all... 😜

![image](https://github.com/devemux86/cruiser/assets/145460147/c83d6879-42d9-452e-926d-5d68ce41ad37)

**Reply by DIrgi** [reply]
Respect, it must be always a hard life for programmers when they ask community for their wishes (I don’t exclude myself)

**Reply by devemux86** [reply]
<img src="https://github.com/devemux86/cruiser/assets/3484020/e9a2db87-0848-4198-bd78-4b754bb8c328" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/3d042fdc-8b55-4d08-b62d-bc6be6fe8aea" width="50%"/>

**Reply by bios4** [reply]
> Respect, it must be always a hard life for programmers when they ask community for their wishes (I don’t exclude myself)

I agree!

This is one of the most valuable things of the app - Emux really takes care to fulfill almost everybody's wishes.

On the other hand: when I get asked about my opinion, I will answer...
If the recipient then likes what I say is up to him. 😉

**Comment by 0709wiwiwi** [comment]
Emux demo.

_It could end up being **very** cramped, transparent panels are recommended
Route diagram plus many top panels do occupy most of the map space.
The compass should probably be placed at the top (mid)position ?
In case of lack of space, present **Left** and **Right** panels ↕ scrollable ?_

**Comment by DolomitesTom** [comment]
Will the long list be scrollable in landscape mode?

**Reply by DIrgi** [reply]
When it becomes scrollable I assume we have missed the target. Either the user has to decide which info he‘d like to display and witch not to do not exceed the available space or if it ends up in scrolling this is even more worse compare to clicking (and switching between displayed data) . Remember you are driving and wear gloves. A click is most times easier the swiping or scrolling in a defined area on the screen.

**Reply by 0709wiwiwi** [reply]
Like @WolfgangZ42 (gloves) tap+toggle @ +% sized panels.
And If set the via point top bar does also shows the distance. 
Or by two simple taps onto **Big Buttons** call up the [Roadbook](https://github.com/devemux86/cruiser/discussions/473)
Scrolling is fine to display many items in easy readable side panels.
Locus uses a scrollable side panel if too many items are inserted. 
Fine for users who don't know (yet) very well what they really need.
Over time, a more minimal set up is probably usually prefered ;-)
Still this.
Cruiser presentation is not only optimised for gloved users  ;-)

**Reply by devemux86** [reply]
About navigation in landscape, we have a discussion, please see my comment there:
- https://github.com/devemux86/cruiser/discussions/242#discussioncomment-9102276

**Reply by 0709wiwiwi** [reply]
> Or by two simple taps onto Big Buttons call up the Roadbook

The [Roadbook ](https://github.com/devemux86/cruiser/discussions/473) **@ active navigation** displays remaining @ arrival time. Cruiser 3.0.43-b5.

**Comment by DIrgi** [comment]
Cool it's implemented and seems working 👍
with the present **cruiser-3.0.43-b9**

**Comment by devemux86** [comment, marked as answer]
Navigation display remaining + arrival time is available in [Cruiser 4.0.0](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
